### PR TITLE
Use homeassistant typing stubs

### PR DESCRIPTION
**Pull request by KapJI**
This is based on beta version of Home Assistant: `2021.4.0b3`. I'll update it when the final version will be released on Wednesday.

This helped to find few bugs:
- The main one is that `async_create_task` should be used instead of `async_add_job`.
- Also `self._client.get_master_token()` may return `Optional[str]`. This wasn't processed correctly.

Part of #85

### CODE REVIEW

File: custom_components/google_home/types.py [hunk 1]
```diff
@@ -1,18 +1,5 @@
 """Various types used in type hints."""
-from typing import Iterable, List, Optional, Set, Tuple, TypedDict
-
-from typing_extensions import Protocol
-
-from homeassistant.helpers.entity import Entity
-
-
-class AddEntitiesCallback(Protocol):
```
Comment (KapJI) [inline custom_components/google_home/types.py:9]: I realised we don't really need such precise type and simple `Callable` is fine for our case.

**Comment by DurgNomis-drol** [comment]
I have seen that someone is going through home-assistant/core to add typing, so maybe your whish is slowly coming true 😊

**Comment by KapJI** [comment]
I think they were only modernising existing types. There are still hundreds of untyped components 🙁

**Comment by DurgNomis-drol** [comment]
Oh okay, we can only hope that in the future this changes 😄

File: custom_components/google_home/__init__.py [hunk 2]
```diff
@@ -30,7 +30,7 @@
 _LOGGER: logging.Logger = logging.getLogger(__package__)
 
 
-async def async_setup(_hass: HomeAssistant, _config: Config) -> bool:
+async def async_setup(_hass: HomeAssistant, _config: dict) -> bool:
```
Comment (KapJI) [inline custom_components/google_home/__init__.py:33]: Btw, this function can be completely removed in 2021.4.0: https://github.com/home-assistant/core/pull/48381

Comment (leikoilja) [reply custom_components/google_home/__init__.py:33]: Nice, let's drop it once HA is on the new version.
If we will do so, shall we specify that `ha-google-home` requires HA `2021.4.0` somewhere (perhaps in `hacs.json` file)?

File: custom_components/google_home/__init__.py [hunk 3]
```diff
@@ -76,7 +76,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     # Offload the loading of entities to the platform
     for platform in PLATFORMS:
-        hass.async_add_job(
+        hass.async_create_task(
```
Comment (leikoilja) [inline custom_components/google_home/__init__.py:79]: Why is this change needed? 
from [docs](http://dev-docs.home-assistant.io/en/master/api/core.html):
> 
> async_add_job(target: Callable[[...], Any], *args: Any) → Optional[_asyncio.Future][source]
> Add a job from within the event loop.
> 
> This method must be run in the event loop.
> 
> target: target to call. args: parameters for method to call.
> 
> async_create_task(target: Coroutine) → _asyncio.Task[source]
> Create a task from within the eventloop.
> 
> This method must be run in the event loop.
> 
> target: target to call.

File: custom_components/google_home/api.py [hunk 4]
```diff
@@ -221,9 +221,9 @@ async def update_google_devices_information(self) -> List[GoogleHomeDevice]:
 
         coordinator_data = await gather(
             *[
-                self.get_alarms_and_timers(device, device.ip_address)
+                self.get_alarms_and_timers(device, device.ip_address, device.auth_token)
                 for device in devices
-                if device.ip_address
+                if device.ip_address and device.auth_token
```
Comment (leikoilja) [inline custom_components/google_home/api.py:226]: Nice catch

File: custom_components/google_home/config_flow.py [hunk 5]
```diff
@@ -54,8 +53,8 @@ async def async_step_user(self, user_input=None):
                 username=user_input[CONF_USERNAME],
                 password=user_input[CONF_PASSWORD],
             )
-            valid, master_token = await self._test_credentials(client)
-            if valid:
+            master_token = await self._test_credentials(client)
+            if master_token is not None:
```
Comment (leikoilja) [inline custom_components/google_home/config_flow.py:57]: Nice catch as well

**Review by leikoilja — COMMENTED** [review]
Nice! Good work with homeassistant stubs as well! 🚀

File: custom_components/google_home/__init__.py (hunk 2 above)
Comment (KapJI) [reply custom_components/google_home/__init__.py:33]: Yes, that's the plan 🙂

File: custom_components/google_home/__init__.py (hunk 3 above)
Comment (KapJI) [reply custom_components/google_home/__init__.py:79]: It doesn't return `Callable`, I checked other places and they use `async_create_task`: https://github.com/home-assistant/core/blob/f4cc4a0896358f4c8df5a193d900716fabb7ad32/homeassistant/components/xiaomi_miio/__init__.py#L126

Comment (leikoilja) [reply custom_components/google_home/__init__.py:79]: Yeh, i guess we had it from [cookiecuter](https://github.com/oncleben31/cookiecutter-homeassistant-custom-component/blob/933ca228d63f5d0c249c9328f16d7f33f6ae1749/%7B%7Bcookiecutter.project_name%7D%7D/custom_components/%7B%7Bcookiecutter.domain_name%7D%7D/__init__.py#L59) where they still use `async_add_job`, but if `async_create_task` is better since it returns `Callable`, as you say, it will probs be good for typing purpose

**Review by leikoilja — APPROVED** [review]
